hazelcast/src/main/java/com/hazelcast/client/impl/protocol/codec/builtin/CustomTypeFactory.java

```diff
@@ -201,13 +201,17 @@ public static EndpointQualifier createEndpointQualifier(int type, String identif
         return EndpointQualifier.resolve(protocolType, identifier);
     }
 
-    public static SqlColumnMetadata createSqlColumnMetadata(String name, int type) {
+    public static SqlColumnMetadata createSqlColumnMetadata(String name, int type, boolean isNullableExists, boolean nullability) {
```

Comment:
Why do we need two flags here (`isNullableExists` and `nullability`) instead of just a straightforward `nullable` that is computed on the upper level?

Reply:
It comes from generated codec (see [here](https://github.com/hazelcast/hazelcast/pull/18114/files#diff-ad1e93b832a69dbc25886c017fa85ce7871da9762ba17dee0e1198dd57d3ca5dR66)).
I also don't realize why it's happens.

Reply:
Got it, thanks.

Reply:
For reference, this `isXXXExists` flag shows whether or not the client sent the data for the `XXX` parameter. For example, for a 4.0 client, `isNullableExists` will be false, and `nullability` will have its default value associated with its type. For 4.2 client, `isNullableExists` will be true and `nullability` will have whatever data sent by the client.

Reply:
@mdumandag thanks, got it.